libcaf_core/caf/config_value.cpp

```diff
@@ -245,14 +245,14 @@ expected<bool> config_value::to_boolean() const {
           if (auto j = x.find("value"); j != x.end()) {
             return j->second.to_boolean();
           }
-          return format_to_error(sec::conversion_failed,
-                                 "missing value for object of type {}", tn);
+          return caf::unexpected{format_to_error(
```

Comment:
We have this a lot now. Let's introduce `format_to_unexpected` to cover this use case with less noise.